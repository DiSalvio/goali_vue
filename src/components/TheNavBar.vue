<template>
  <div v-if="token" class="navbar px-0 py-2 mb-3 navbar-border">
    <button id="back-button" @click="$router.back()" class="btn btn-secondary ml-2">
      <font-awesome-icon icon="backward-step" />
      Back
    </button>
    <button id="home-button" @click="$router.push({name: 'PageHome'})" class="btn btn-success text-white">
      <img src="../assets/hiking.png">
      Home
      <img src="../assets/mission-medium.png">
    </button>
    <button id="log-out-button" @click="logOut" class="btn btn-primary mr-2">
      Log Out
      <font-awesome-icon icon="door-open" />
    </button>
  </div>
</template>

<script>
export default {
  computed: {
    token () {
      return this.$store.state.userModule.token
    }
  },
  methods: {
    async logOut () {
      const loggedOut = await this.$store.dispatch('logOut', this.token)
      if (loggedOut) {
        this.$router.push({name: 'PageLogin'})
      }
    }
  },
  async created () {
    if (localStorage.getItem('token')) {
      await this.$store.dispatch('refreshTokenState', localStorage.getItem('token'))
    }
  }
}
</script>

<style>
.navbar-border {
  border: 1px solid rgba(0, 0, 0, 0.125);
}
</style>
