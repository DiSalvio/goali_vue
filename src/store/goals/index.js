import {
  findById,
  activeOnly,
  completedOnly,
  inProgressOnly,
  updateItemInArray,
  urlHelper,
  authHeader
} from '@/helpers/index.js'
import axios from 'axios'

const goalModule = {
  state () {
    return {
      goals: []
    }
  },
  getters: {
    goal (state) {
      return (goalId) => findById(state.goals, goalId)
    },
    activeGoals (state) {
      return activeOnly(state.goals)
    },
    activeCompletedGoals (state, getters) {
      return completedOnly(getters.activeGoals)
    },
    activeInProgressGoals (state, getters) {
      return inProgressOnly(getters.activeGoals)
    }
  },
  actions: {
    async fetchGoals ({ commit }, token) {
      return await axios.get(urlHelper({ resource: 'goals' }), authHeader(token))
        .then((response) => {
          commit('setGoals', response.data)
        })
        .catch((e) => {
          console.log(e)
        })
    },
    async fetchGoal ({ dispatch }, { goalId, token }) {
      return await axios.get(urlHelper({ ids: [ goalId ]}), authHeader(token))
        .then((response) => {
          dispatch('upsertGoal', response.data)
          return response.data
        })
        .catch((e) => {
          console.log(e)
          return null
        })
    },
    async createGoal ({ commit }, { name, description, token }) {
      return await axios.post(urlHelper({ resource: 'goals' }), {
        name,
        description,
        completed: false,
        updated: new Date(Date.now()).toISOString(),
        timestamp: new Date(Date.now()).toISOString(),
        removed: false
      }, 
        authHeader(token)
      )
        .then((response) => {
          commit('pushGoal',  response.data)
          return response.data
        })
        .catch((error) => {
          console.log(error)
        })
    },
    async saveEditedGoal ({ commit }, { editedGoal, token }) {
      return await axios.put(urlHelper({ ids: [editedGoal.id] }), {
        ...editedGoal,
        updated: new Date(Date.now()).toISOString()
      },
        authHeader(token)
      )
        .then((response) => {
          commit('updateGoal', { item: response.data })
          return true
        })
        .catch((error) => {
          console.log(error)
          return false
        })
    },
    upsertGoal ({ state, commit }, goal) {
      if (findById(state.goals, goal)) {
        commit('updateGoal', { item: goal })
      } else {
        commit('pushGoal', goal)
      }
    }
  },
  mutations: {
    setGoals (state, goals) {
      state.goals = goals
    },
    pushGoal (state, newGoal) {
      state.goals.push(newGoal)
    },
    updateGoal: updateItemInArray({ resource: 'goals' })
  }
}

export default goalModule
