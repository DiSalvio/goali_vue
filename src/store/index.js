import { createStore } from 'vuex'
import userModule from './users/index.js'
import goalModule from './goals/index.js'
import taskModule from './tasks/index.js'
import subTaskModule from './subtasks/index.js'
import sourceData from '@/data.json'

const store = createStore({
  state: {
    users: sourceData.users,
    goals: sourceData.goals,
    tasks: sourceData.tasks,
    subTasks: sourceData.subTasks
  },
  modules: {
    userModule,
    goalModule,
    taskModule,
    subTaskModule
  }
})

export default store
