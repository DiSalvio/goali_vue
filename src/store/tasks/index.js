import getters from '@/store/tasks/getters.js'
import actions from '@/store/tasks/actions.js'
import mutations from '@/store/tasks/mutations.js'

const taskModule = {
  state () {
    return {
      tasks: []
    }
  },
  getters,
  actions,
  mutations 
}

export default taskModule
