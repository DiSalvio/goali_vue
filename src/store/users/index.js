const userModule = {
  state () {
    return {
    }
  },
  getters: {
  },
  actions: {
  },
  mutations: {
  }
}

export default userModule
