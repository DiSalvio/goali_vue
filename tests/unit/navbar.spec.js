import store from '@/store'
import { mount } from '@vue/test-utils'
import TheNavBar from '@/components/TheNavBar.vue'

describe('TheNavBar.vue', () => {

  describe('default set up', () => {
    let wrapper

    beforeEach(() => {
      wrapper = mount(TheNavBar, {
        global: {
          plugins: [store]
        }
      })
    })

    it('has token computed prop and logOut method', () => {
      expect(wrapper.vm).toEqual(
        expect.objectContaining({
          'token': null,
          'logOut': expect.any(Function)
        })
      )
    })

  })

  describe('when not logged in', () => {
    let wrapper

    beforeEach(() => {
      wrapper = mount(TheNavBar, {
        global: {
          plugins: [store]
        }
      })
    })

    it('does not have token set in localStorage or computed prop', () => {
      expect(localStorage.getItem('token')).toBe(null)
      expect(wrapper.vm.token).toBe(null)
    })

    it('back, home, and log out buttons are not shown', () => {
      const buttons = wrapper.findAll('button')
      expect(buttons.length).toBe(0)
    })

    it('does not dispatch refresh token api call', () => {
      const { vm } = mount(TheNavBar, {
        global: {
          mocks: {
            $store: {
              state: {
                userModule: {
                  token: null
                }
              },
              dispatch: jest.fn()
            }
          }
        }
      })

      expect(vm.$store.dispatch).not.toHaveBeenCalled()
    })
  })

  describe('when logged in', () => {
    let wrapper

    beforeEach(() => {
      localStorage.setItem('token', '1234')
      wrapper = mount(TheNavBar, {
        global: {
          mocks: {
            $store: {
              state: {
                userModule: {
                  token: '1234'
                }
              },
              dispatch: jest.fn()
            }
          }
        }
      })
    })

    it('has token set in computed prop', () => {
      expect(wrapper.vm.token).toBe('1234')
    })

    it('dispatches refresh token action with correct params', () => {
      expect(wrapper.vm.$store.dispatch).toHaveBeenCalledWith('refreshTokenState', '1234')
    })

    it('shows back, home, and log out buttons in nav bar', () => {
      const buttons = wrapper.findAll('button')
      expect(buttons.length).toBe(3)
      expect(buttons[0].attributes('id')).toBe('back-button')
      expect(buttons[1].attributes('id')).toBe('home-button')
      expect(buttons[2].attributes('id')).toBe('log-out-button')
    })

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('logOut method', () => {
    let wrapper

    beforeEach(() => {
      localStorage.setItem('token', '5678')
      wrapper = mount(TheNavBar, {
        global: {
          mocks: {
            $route: {
              name: 'PageLogin'
            },
            $router: {
              push: jest.fn()
            },
            $store: {
              state: {
                userModule: {
                  token: '5678'
                }
              },
              dispatch: jest.fn().mockResolvedValue(true)
            }
          }
        }
      })
    })
  
    it('correctly dispatches logOut action with correct params', async () => {
      await wrapper.vm.logOut()
      expect(wrapper.vm.$store.dispatch).toHaveBeenCalledWith('logOut', '5678')
    })

    it('redirects to login page if logOut is successful', async () => {
      await wrapper.vm.logOut()
      expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'PageLogin' })
    })
  })

  describe('user', () => {
    let wrapper

    beforeEach(() => {
      localStorage.setItem('token', '5678')
      wrapper = mount(TheNavBar, {
        global: {
          mocks: {
            $route: {
              name: 'PageHome'
            },
            $router: {
              push: jest.fn(),
              back: jest.fn()
            },
            $store: {
              state: {
                userModule: {
                  token: '5678'
                }
              },
              dispatch: jest.fn().mockResolvedValue(true)
            },
          }
        }
      })
    })

    it('can push back button to go back', () => {
      const backButton = wrapper.find('#back-button')
      backButton.trigger('click')
      expect(wrapper.vm.$router.back).toHaveBeenCalled()
    })

    it('can push home button and be routed to home page', async () => {
      const homeButton = wrapper.find('#home-button')
      await homeButton.trigger('click')
      expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'PageHome' })
    })

    it('can push log out button and be logged out', async () => {
      const logOutSpy = jest.spyOn(wrapper.vm, 'logOut')

      let buttons = wrapper.findAll('button')
      expect(buttons.length).toBe(3)

      const logOutButton = wrapper.find('#log-out-button')
      await logOutButton.trigger('click')
      expect(logOutSpy).toHaveBeenCalled()
      expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'PageLogin' })
    })

  })

})
