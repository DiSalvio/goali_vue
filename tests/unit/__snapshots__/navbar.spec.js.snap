// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavBar.vue when logged in matches snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="navbar px-0 py-2 mb-3 navbar-border"
      >
        <button
          class="btn btn-secondary ml-2"
          id="back-button"
        >
          <font-awesome-icon
            icon="backward-step"
          />
           Back 
        </button>
        <button
          class="btn btn-success text-white"
          id="home-button"
        >
          <img
            src=""
          />
           Home 
          <img
            src=""
          />
        </button>
        <button
          class="btn btn-primary mr-2"
          id="log-out-button"
        >
           Log Out 
          <font-awesome-icon
            icon="door-open"
          />
        </button>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "$store": Object {
          "dispatch": [MockFunction] {
            "calls": Array [
              Array [
                "refreshTokenState",
                "1234",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "state": Object {
            "userModule": Object {
              "token": "1234",
            },
          },
        },
      },
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "$store": Object {
          "dispatch": [MockFunction] {
            "calls": Array [
              Array [
                "refreshTokenState",
                "1234",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "state": Object {
            "userModule": Object {
              "token": "1234",
            },
          },
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "$store": Object {
            "dispatch": [MockFunction] {
              "calls": Array [
                Array [
                  "refreshTokenState",
                  "1234",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "state": Object {
              "userModule": Object {
                "token": "1234",
              },
            },
          },
          "hasOwnProperty": [Function],
          "logOut": [Function],
          "token": "1234",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "$store": 4,
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "logOut": 4,
            "toJSON": 0,
            "token": 4,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "$store": Object {
              "dispatch": [MockFunction] {
                "calls": Array [
                  Array [
                    "refreshTokenState",
                    "1234",
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
              "state": Object {
                "userModule": Object {
                  "token": "1234",
                },
              },
            },
            "hasOwnProperty": [Function],
            "logOut": [Function],
            "token": "1234",
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "$store": Object {
              "dispatch": [MockFunction] {
                "calls": Array [
                  Array [
                    "refreshTokenState",
                    "1234",
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
              "state": Object {
                "userModule": Object {
                  "token": "1234",
                },
              },
            },
            "hasOwnProperty": [Function],
            "logOut": [Function],
            "token": "1234",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": false,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": "1234",
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": Array [
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": Array [],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <font-awesome-icon
                      icon="backward-step"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "icon": "backward-step",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "font-awesome-icon",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " Back ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el":  Back ,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  class="btn btn-secondary ml-2"
                  id="back-button"
                >
                  <font-awesome-icon
                    icon="backward-step"
                  />
                   Back 
                </button>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "btn btn-secondary ml-2",
                  "id": "back-button",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <img
                      src=""
                    />,
                    "key": null,
                    "patchFlag": -1,
                    "props": Object {
                      "src": "",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "img",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " Home ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el":  Home ,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <img
                      src=""
                    />,
                    "key": null,
                    "patchFlag": -1,
                    "props": Object {
                      "src": "",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "img",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  class="btn btn-success text-white"
                  id="home-button"
                >
                  <img
                    src=""
                  />
                   Home 
                  <img
                    src=""
                  />
                </button>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "btn btn-success text-white",
                  "id": "home-button",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " Log Out ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el":  Log Out ,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <font-awesome-icon
                      icon="door-open"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "icon": "door-open",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "font-awesome-icon",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  class="btn btn-primary mr-2"
                  id="log-out-button"
                >
                   Log Out 
                  <font-awesome-icon
                    icon="door-open"
                  />
                </button>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "btn btn-primary mr-2",
                  "id": "log-out-button",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [],
            "dynamicProps": null,
            "el": <div
              class="navbar px-0 py-2 mb-3 navbar-border"
            >
              <button
                class="btn btn-secondary ml-2"
                id="back-button"
              >
                <font-awesome-icon
                  icon="backward-step"
                />
                 Back 
              </button>
              <button
                class="btn btn-success text-white"
                id="home-button"
              >
                <img
                  src=""
                />
                 Home 
                <img
                  src=""
                />
              </button>
              <button
                class="btn btn-primary mr-2"
                id="log-out-button"
              >
                 Log Out 
                <font-awesome-icon
                  icon="door-open"
                />
              </button>
            </div>,
            "key": 0,
            "patchFlag": 0,
            "props": Object {
              "class": "navbar px-0 py-2 mb-3 navbar-border",
              "key": 0,
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "computed": Object {
              "token": [Function],
            },
            "created": [Function],
            "methods": Object {
              "logOut": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 17,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="navbar px-0 py-2 mb-3 navbar-border"
        >
          <button
            class="btn btn-secondary ml-2"
            id="back-button"
          >
            <font-awesome-icon
              icon="backward-step"
            />
             Back 
          </button>
          <button
            class="btn btn-success text-white"
            id="home-button"
          >
            <img
              src=""
            />
             Home 
            <img
              src=""
            />
          </button>
          <button
            class="btn btn-primary mr-2"
            id="log-out-button"
          >
             Log Out 
            <font-awesome-icon
              icon="door-open"
            />
          </button>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "computed": Object {
            "token": [Function],
          },
          "created": [Function],
          "methods": Object {
            "logOut": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 16,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="navbar px-0 py-2 mb-3 navbar-border"
        >
          <button
            class="btn btn-secondary ml-2"
            id="back-button"
          >
            <font-awesome-icon
              icon="backward-step"
            />
             Back 
          </button>
          <button
            class="btn btn-success text-white"
            id="home-button"
          >
            <img
              src=""
            />
             Home 
            <img
              src=""
            />
          </button>
          <button
            class="btn btn-primary mr-2"
            id="log-out-button"
          >
             Log Out 
            <font-awesome-icon
              icon="door-open"
            />
          </button>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 8,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "$store": Object {
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            "refreshTokenState",
            "1234",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "state": Object {
        "userModule": Object {
          "token": "1234",
        },
      },
    },
    "hasOwnProperty": [Function],
    "logOut": [Function],
    "token": "1234",
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "$store": Object {
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            "refreshTokenState",
            "1234",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "state": Object {
        "userModule": Object {
          "token": "1234",
        },
      },
    },
  },
  "wrapperElement": <div
    class="navbar px-0 py-2 mb-3 navbar-border"
  >
    <button
      class="btn btn-secondary ml-2"
      id="back-button"
    >
      <font-awesome-icon
        icon="backward-step"
      />
       Back 
    </button>
    <button
      class="btn btn-success text-white"
      id="home-button"
    >
      <img
        src=""
      />
       Home 
      <img
        src=""
      />
    </button>
    <button
      class="btn btn-primary mr-2"
      id="log-out-button"
    >
       Log Out 
      <font-awesome-icon
        icon="door-open"
      />
    </button>
  </div>,
}
`;
